### PR TITLE
[Core] Add CDI2 integration

### PR DESCRIPTION
**Pull request by rmannibucau**
## Summary

OpenWebBeans PR looks outdated now CDI 2 has a dedicated API to manage a container.
Note it also unify weld module so it can likely be removed after some deprecation time.
To be complete the deltaspike module should stay a bit for CDI 1.x containers.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: cdi2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    </parent>
+
+    <artifactId>cucumber-cdi2</artifactId>
+    <packaging>jar</packaging>
+    <name>Cucumber-JVM: CDI 2</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.cucumber</groupId>
+            <artifactId>cucumber-java</artifactId>
```
Comment (mpkorstanje) [inline cdi2/pom.xml]: I've just merged https://github.com/cucumber/cucumber-jvm/pull/1615. By implementing `io.cucumber.core.backend.ObjectFactory` only a dependency on `cucumber-core` is needed.

File: cdi2/src/main/java/cucumber/api/cdi2/README.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package cucumber.api.cdi2;
+
+public class README {
```
Comment (mpkorstanje) [inline cdi2/src/main/java/cucumber/api/cdi2/README.java]: What does this do?

**Comment by mpkorstanje** [comment]
Looks like the build is failing on Java 11. This can be solved by adding:

```
    <groupId>javax.annotation</groupId>
    <artifactId>javax.annotation-api</artifactId>
```

**Comment by rmannibucau** [comment]
@mpkorstanje just pushed a commit, hopefully fixing all comments - and thanks a lot for the catches up!

**Comment by aslakhellesoy** [comment]
Hi @rmannibucau,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by mpkorstanje** [comment]
Cheers!

**Comment by aslakhellesoy** [comment]
Hi @rmannibucau,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**